internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,211 @@
+
+        mkdir -p controlplane/etcd
+        echo "Extracting etcd /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane etcd > controlplane/etcd/metrics
+
+        mkdir -p controlplane/controller-manager
+        echo "Extracting controller-manager /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane controllermanager > controlplane/controller-manager/metrics
+
+        mkdir -p controlplane/scheduler
+        echo "Extracting scheduler /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane scheduler > controlplane/scheduler/metrics
+    else
+        echo "Skipping control plane metrics"
+    fi
+
+    # K8s objects
+    echo "Generating list of kubernetes resources"
+    kubedump namespaces
+    kubedump services
+    kubedump pods
+
+    cd -
+    cleanup
+}
+
+# bootstrap suvcommand installs the required components in the cluster to generate the testdata.
+# If $SKIP_INSTALL is non-empty, it skips deploying KSM, the dummy resources, and the scrapper pod and just copies
+# script inside an scraper pod that is assumed to exist already.
+function bootstrap() {
+    echo -e "Using context $(kubectl config current-context), is this ok?\n^C now if it is not."
+    read
+
+    if [[ -z $SKIP_INSTALL ]]; then
+        echo "Installing e2e-resources chart"
+        helm dependency update ../../../e2e/charts/e2e-resources > /dev/null
+        helm upgrade --install e2e ../../../e2e/charts/e2e-resources -n mock --create-namespace
+
+        echo "Installing KSM"
+        helm dependency update ./deployments/ksm > /dev/null
+        helm upgrade --install ksm ./deployments/ksm -n ksm --create-namespace --wait
+
+        echo "Deploying scraper pods"
+        kubectl apply -f ./deployments/scraper.yaml
+    fi
+
+    echo "Waiting for scraper pod to be ready"
+    kubectl -n scraper wait --for=condition=Ready pod -l "$scrapper_selector"
+    pod=$(scraper_pod "$scrapper_selector")
+    if [[ -z "pod" ]]; then
+        echo "Could not find scraper pod (-l $scrapper_selector)"
+        exit 1
+    fi
+    echo "Found scraper pod $pod"
+
+    echo "Copying datagen.sh to scraper pods"
+    kubectl cp datagen.sh scraper/$pod:/bin/
+}
+
+# cleanup uninstalls the dummy resources and the scraper pod.
+function cleanup() {
+    echo "Removing e2e-resources chart"
+    helm uninstall e2e -n mock || true
+    echo "Removing ksm"
+    helm uninstall ksm -n ksm || true
+    echo "Removing scraper pods"
+    kubectl delete -f ./deployments/scraper.yaml --wait || true
+}
+
+# scrape will curl the specified component and output the response body to standard output.
+function scrape() {
+    bearer="Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+    endpoint=unsupported://
+
+    case $1 in
+    ksm)
+      endpoint=http://ksm-kube-state-metrics.ksm.svc:8080/metrics
```

Comment:
It would be great if these could be overridden from env vars, like `KSM_ENDPOINT=${KSM_ENDPOINT:-http://ksm-kube-state-metrics.ksm.svc:8080/metrics}`

Reply:
Done